### DISCUSSION TITLE
Newbie question: very first commit failed due to typo in password. How do I fix this?

### DISCUSSION THREAD

**Discussion by HughCan** [community/community#21842] [New to GitHub]
<p>I am rather new to Git/GitHub so I only have a basic knowledge of the basic concepts. I am developing some vue.js code in Visual Studio Code and trying to push my code to the remote repository on GitHub. (I *think* I have successfully staged and added my code to the local repository.) When I did my very first push of my code to a new GitHub repository, I believe I made a typo in my password and the push failed on an authentication error. That seems pretty reasonable to me. I assumed that simply trying the push again would give me another chance to type the password, get it right this time, then do the push successfully but that isn’t what happened. The Git log doesn’t want to let me push and thinks I might need to do a pull, which doesn’t make a lot of sense to me because this is the very first push to master, in a new GitHub repository and I am the only person developing on this little project.</p>
<p>I have no idea how to proceed and finish my push. Can anyone advise me?</p>
<p>For what it’s worth, I tried a pull even though it didn’t seem appropriate but it didn’t work and my subsequent attempt to push failed too.</p>
<p>Here’s the relevant bit of the Git log:</p>
<pre><code>&gt; git push -u vue_crash_todolist master
Logon failed, use ctrl+c to cancel basic credential prompt.
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/HughCan/vue_crash_todolist/'
&gt; git push -u vue_crash_todolist master
To https://github.com/HughCan/vue_crash_todolist
 ! [rejected] master -&gt; master (fetch first)
error: failed to push some refs to 'https://github.com/HughCan/vue_crash_todolist'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
&gt; git pull --tags
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull &lt;remote&gt; &lt;branch&gt;

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=vue_crash_todolist/&lt;branch&gt; master
&gt; git status -z -u
&gt; git symbolic-ref --short HEAD
&gt; git rev-parse master
&gt; git rev-parse --symbolic-full-name master@{u}
fatal: no upstream configured for branch 'master'
&gt; git for-each-ref --format %(refname) %(objectname) --sort -committerdate
&gt; git remote --verbose
&gt; git config --get commit.template
&gt; git push -u vue_crash_todolist master
To https://github.com/HughCan/vue_crash_todolist
 ! [rejected] master -&gt; master (fetch first)
error: failed to push some refs to 'https://github.com/HughCan/vue_crash_todolist'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
&gt; git push -u vue_crash_todolist master
To https://github.com/HughCan/vue_crash_todolist
 ! [rejected] master -&gt; master (fetch first)
error: failed to push some refs to 'https://github.com/HughCan/vue_crash_todolist'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
</code></pre>
<p> </p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/hughcan">@hughcan</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<p>Ok I try to help understand what is going on, let’s take the first two commands:</p>
<blockquote>
<blockquote>
<p>git push -u vue_crash_todolist master<br>
Logon failed, use ctrl+c to cancel basic credential prompt.<br>
remote: Invalid username or password.<br>
fatal: Authentication failed for ‘<a href="https://github.com/HughCan/vue_crash_todolist/" class="inline-onebox-loading" rel="nofollow noopener">https://github.com/HughCan/vue_crash_todolist/</a>’<br>
git push -u vue_crash_todolist master<br>
To <a href="https://github.com/HughCan/vue_crash_todolist" class="inline-onebox-loading" rel="nofollow noopener">https://github.com/HughCan/vue_crash_todolist</a><br>
! [rejected] master -&gt; master (fetch first)<br>
error: failed to push some refs to ‘<a href="https://github.com/HughCan/vue_crash_todolist" class="inline-onebox-loading" rel="nofollow noopener">https://github.com/HughCan/vue_crash_todolist</a>’<br>
hint: Updates were rejected because the remote contains work that you do<br>
hint: not have locally. This is usually caused by another repository pushing<br>
hint: to the same ref. You may want to first integrate the remote changes<br>
hint: (e.g., ‘git pull …’) before pushing again.<br>
hint: See the ‘Note about fast-forwards’ in ‘git push --help’ for details.</p>
</blockquote>
</blockquote>
<p>When you fail to authenticate there is no problem, so you try another time and it give you an error and reject the push to master. This can happend if there are some commits in the branch on remote that you don’t have in your local machine and I think this is what is going on. There is a README.md file in your remote repository that you don’t have in the local repository.</p>
<p>if you want to be sure you don’t lose any commits done locally, you can first create a new branch using this command:</p>
<pre><code>git branch new_branch
</code></pre>
<p>Using branching you can safely put somewhere what you have done and not loose anything.</p>
<p>Then run this command:</p>
<pre><code>git pull origin master
</code></pre>
<p>The pull command need the remote and the branch from where to pull, in your case the master branch is located in the origin remote on github.</p>
<p>Can happend 2 things: the pull will be rejected or the pull use a merge strategy to align the master branch on you local machine. In the first case the procedure can be ugly, but we can do anything using git and some command. If everything goes ok, you can now push like you tried to do previously and remove the branch created with the branch command:</p>
<pre><code>git branch --delete new_branch
</code></pre>
<p>Let me know if this solve the problem :slight_smile:</p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with<span> </span>Accept as Solution<span> </span>to help other users locate important info. Don’t forget to give<span> </span>Kudos<span> </span>for great contents!</p>
